### PR TITLE
Fix expiry process for version meta data files

### PR DESCRIPTION
**Pull request by JammingBen**
## Description
This fixes the massive amount of error logs being produced during this.

## Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Database schema changes (next release will require increase of minor version instead of patch)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Technical debt
- [ ] Tests only (no source changes)

### CODE REVIEW

**Comment by JammingBen** [comment]
> If I understand correctly, you're skipping the metadata files while getting the versions, but I assume that the metadata will be removed when the version is also removed.

Exactly, sorry, the title is a bit misleading. Version meta files get removed eventually if the corresponding versions get removed.

**Review by jvillafanez — APPROVED** [review]

File: apps/files_versions/lib/FileHelper.php [hunk 1]
```diff
@@ -82,6 +82,11 @@ public function getAllVersions($uid) {
 			$files = $view->getDirectoryContent($dir);
 			foreach ($files as $file) {
 				$filePath = $dir . '/' . $file->getName();
+				$isMetaFile = \substr($filePath, -\strlen(MetaStorage::VERSION_FILE_EXT)) === MetaStorage::VERSION_FILE_EXT;
```
Comment (phil-davis) [inline apps/files_versions/lib/FileHelper.php:85]: This should work, even when the real file is `something.json`. I tried by creating `f.json` and editing it a few times. In `data/username/files_versions` I get:
```
$ ls -l f*
-rw-rw-r-- 1 phil phil 60 Aug 31 15:12 f.json.current.json
-rw-rw-r-- 1 phil phil 12 Aug 31 15:11 f.json.v1661938001
-rw-rw-r-- 1 phil phil 60 Aug 31 15:11 f.json.v1661938001.json
-rw-rw-r-- 1 phil phil 13 Aug 31 15:12 f.json.v1661938002
-rw-rw-r-- 1 phil phil 60 Aug 31 15:11 f.json.v1661938002.json
```
The real files end up called `f.json.vnnnnnnnnnn` and so their file names do not end in `.json` - good.

**Review by phil-davis — APPROVED** [review]

**Comment by phil-davis** [comment]
@jnweiger - should we switch this over to be  PR to `release-10.11.0` or merge this and cherry-pick to the release branch?

**Comment by jnweiger** [comment]
One-flow says master first, but the simpler process is release branch first, I am fine with both.

**Comment by JammingBen** [comment]
Then let's do it the easy way and merge it into `release-10.11.0` :)

**Pull request merged**